hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -595,9 +595,16 @@ private void doInvokeLocal(boolean isAsync) {
     private void doInvokeRemote() {
         assert connectionManager != null : "Endpoint manager was null";
 
-        ServerConnection connection = connectionManager.getOrConnect(targetAddress);
+        ServerConnection connection = connectionManager.getOrConnect(targetAddress, op.getPartitionId());
```

Comment:
Will `streamId` always be the `partitionId`? These are different concerns though, but it feels (so far) we just introduce a different term for the same thing. The property talks about channel count, its description partition count. So just a reminder for the final version: we need to agree on the term and use it consistently.

Reply:
Channel count is not partition count. Channel count will be very low; partition count is much higher.

About partition id and stream id; n this PR it is the same thing although partition id -1 will always be mapped to stream 0. But streams could be used for other things that want to guarantee ordering within the stream.

Reply:
Also partitions can move around. So it could be that one moment your partition is one one machine and one moment later on another. But streams do not move around; streams are within a connection.